tickit/devices/eiger/eiger.py

```diff
@@ -0,0 +1,453 @@
+                    "series": "<series id>",
+                    "frame": aquired.index,
+                    "hash": aquired.hash,
+                }
+
+                json2 = {
+                    "htype": "dimage_d-1.0",
+                    "shape": "[x,y,(z)]",
+                    "type": aquired.dtype,
+                    "encoding": aquired.encoding,
+                    "size": len(aquired.data),
+                }
+
+                json3 = {
+                    "htype": "dconfig-1.0",
+                    "start_time": "<start_time>",
+                    "stop_time": "<stop_time>",
+                    "real_time": "<real_time>",
+                }
+
+                LOGGER.debug(header_json)
+                LOGGER.debug(json2)
+                LOGGER.debug(json3)
+
+            return "Aquiring Data from Eiger..."
+        else:
+            return (
+                f"Ignoring trigger, state={self.status.state},"
+                f"trigger_mode={trigger_mode}"
+            )
+
+    async def cancel(self) -> None:
+        """Function to stop the data acquisition.
+
+        Function to stop the data acquisition, but only after the next
+        image is finished.
+        """
+        self._set_state(State.READY)
+
+        header_json = {"htype": "dseries_end-1.0", "series": "<id>"}
+
+        LOGGER.debug(header_json)
+
+    async def abort(self) -> None:
+        """Function to abort the current task on the Eiger."""
+        self._set_state(State.IDLE)
+
+        header_json = {"htype": "dseries_end-1.0", "series": "<id>"}
+
+        LOGGER.debug(header_json)
+
+    def update(self, time: SimTime, inputs) -> DeviceUpdate:
+        """Generic update function to update the values of the ExampleHTTPDevice.
+
+        Args:
+            time (SimTime): The simulation time in nanoseconds.
+            inputs (Inputs): A TypedDict of the inputs to the ExampleHTTPDevice.
+
+        Returns:
+            DeviceUpdate[Outputs]:
+                The produced update event which contains the value of the device
+                variables.
+        """
+        current_flux = inputs["flux"]
+        # if current_flux > 0:
+        # TODO: Figure out how to scale the output image
+        intensity_scale = (current_flux / 100) * 100
+        LOGGER.debug(f"Relative beam intensity: {intensity_scale}")
+
+        return DeviceUpdate(self.Outputs(), SimTime(time + int(1e8)))
+
+    def get_state(self) -> Value:
+        """Returns the current state of the Eiger.
+
+        Returns:
+            State: The state of the Eiger.
+        """
+        val = self.status.state
+        allowed = [s.value for s in State]
+        return serialize(
+            Value(
+                val,
+                AccessMode.STRING,
+                access_mode=AccessMode.READ_ONLY,
+                allowed_values=allowed,
+            )
+        )
+
+    def _set_state(self, state: State):
+        self.status.state = state
+
+
+class EigerRESTAdapter(
```

Comment:
Think I would be tempted to split this file into two (`eiger_device` & `eiger_adapter`)

Reply:
@callumforrester what do you think?

Reply:
Both classes are going to get big so maybe. I don't think it matters too much though, whatever you prefer.

Reply:
Should I put the adapter in the `tickit/adapters` dir or in with the other Eiger scripts?

Reply:
Put all the Eiger stuff together